java/c/src/test/java/org/apache/arrow/c/RoundtripTest.java

```diff
@@ -951,6 +952,86 @@ public void testImportReleasedArray() {
     }
   }
 
+  @Test
```

Comment:
@lidavidm would this test be satisfactory to make sure we make the sliced vector manually in Java and test C data interface?

Reply:
Is this right? A sliced vector should be exported with the same buffers as a normal vector, but should have an offset applied. The actual buffers should not be changed

Reply:
Right, one question related to that. From C++ we could get the full offset, validity and data buffers, that's right. In Java, we would use the `arrowArrayOffset` to determine the first offset, and take dataLength + 1 from the first offset. This is a must right? 

And for validity buffer we can just import what is sent, and for data buffer, we need to determine the capacity, because the known information are via the adjustedOffset buffer. So that last value would have the length of the buffer we need to read. Though this will contain unnecessary values at the beginning of the buffer, but since we have correct offsets, the vector can interpret the data accurately. 

Is this understanding correct? Or does it have issues?

Reply:
Vector doesn't understand offsets. So all buffers have to be sliced. Also, validity buffer and offsets buffer need to be copied, because (1) you can't slice on a bit boundary and (2) the offsets need to be fixed

Reply:
Right. I see your point. This means we need to add the offset to the `ArrowArray.Snapshot` rather than creating the vector like this. I see.

Reply:
I'm still not sure about the tests here, I think the way we should do it is by manually tweaking the Snapshot instance